docs/architecture/engine.md

```diff
@@ -0,0 +1,197 @@
+   │     │    ┌──────────────┤  Shard 1  ├──────────────┐     │      │
+   │     │    │              └───────────┘              │     │      │
+   │     │    │                                         │     │      │
+   │     │    │                                         │     │      │
+   │     │    │              ┌───────────┐              │     │      │
+   │     │    │   ┌──────────┤ Series 1  ├──────────┐   │     │      │
+   │     │    │   │          └───────────┘          │   │     │      │
+   │     │    │   │                                 │   │     │      │
+   │     │    │   │                                 │   │     │      │
+   │     │    │   │ ┌─────────────────────────────┐ │   │     │      │
+   │     │    │   │ │      Block [2PM - 4PM]      │ │   │     │      │
+   │     │    │   │ ├─────────────────────────────┤ │   │     │      │
+   │     │    │   │ │      Block [4PM - 6PM]      │ │   │     │      │
+   │     │    │   │ ├─────────────────────────────┤ │   │     │      │
+   │     │    │   │ │       ┌────────────┐        │ │   │     │      │
+   │     │    │   │ └───────┤   Blocks   ├────────┘ │   │     │      │
+   │     │    │   │         └────────────┘          │   │     │      │
+   │     │    │   │                                 │   │     │      │
+   │     │    │   │                                 │   │     │      │
+   │     │    │   │  ┌────────────────────────────┐ │   │     │      │
+   │     │    │   │  │                            │ │   │     │      │
+   │     │    │   │  │     Block [6PM - 8PM]      │ │   │     │      │
+   │     │    │   │  │                            │ │   │     │      │
+   │     │    │   │  ├────────────────────────────┤ │   │     │      │
+   │     │    │   │  │ Active Buffers (encoders)  │ │   │     │      │
+   │     │    │   │  └────────────────────────────┘ │   │     │      │
+   │     │    │   │                                 │   │     │      │
+   │     │    │   │                                 │   │     │      │
+   │     │    │   └─────────────────────────────────┘   │     │      │
+   │     │    │                                         │     │      │
+   │     │    │                                         │     │      │
+   │     │    │                                         │     │      │
+   │     │    │                                         │     │      │
+   │     │    └─────────────────────────────────────────┘     │      │
+   │     │                                                    │      │
+   │     │                                                    │      │
+   │     └────────────────────────────────────────────────────┘      │
+   │                                                                 │
+   └─────────────────────────────────────────────────────────────────┘
+```
+
+The in-memory portion of M3DB is implemented via a hierarchy of objects:
+
+1. A "database" of which there is only one per M3DB process.
```

Comment:
I don't like the use of quotes for "database", reads like air quotes to me. Would you mind switching to block quotes, i.e. `database` instead?

Same for all the types in the sections below

Reply:
done